components/camel-cdi/src/main/java/org/apache/camel/cdi/CdiCamelExtension.java

```diff
@@ -373,6 +375,11 @@ private boolean shouldDeployDefaultCamelContext(Set<Bean<?>> beans) {
     }
 
     private void afterDeploymentValidation(@Observes AfterDeploymentValidation adv, BeanManager manager) {
+    	if(!configuration.isUnmodifiable()) {
```

Comment:
At that stage the event is always modifiable, right? So the check could be removed?